crates/hir-def/src/nameres/path_resolution.rs

```diff
@@ -470,9 +472,9 @@ impl DefMap {
         };
 
         let extern_prelude = || {
-            if self.block.is_some() {
-                // Don't resolve extern prelude in block `DefMap`s, defer it to the crate def map so
-                // that blocks can properly shadow them
+            if matches!(self[module].origin, ModuleOrigin::BlockExpr { .. }) {
```

Comment:
```suggestion
            if self.block.is_some() && module == DefMap::ROOT {
```

If I understand the code here correctly, likjewise below. Though I am unsure whether this is the correct fix here

Reply:
Though it should be more correct than the status quo

Reply:
They are equivalent. If a module is a `BlockExpr`, its `local_id` should be equivalent to `ROOT`, and it should be within a block; conversely, this is also true. Should I change it to the latter?

Reply:
Yes, sorry I wasn't clear. With unsure whether this is the correct fix I meant the general approach here. My code suggestion is the more correct way to do your check here (they are equivalent, it's just that the origin shouldn't be used to check this property).